Source/ARTRealtime.m

```diff
@@ -212,10 +215,27 @@ - (instancetype)initWithOptions:(ARTClientOptions *)options {
         if (options.autoConnect) {
             [self _connect];
         }
+        
+        if (options.recover) { //RTN16f
+            ARTConnectionRecoveryKey *recoveryKey = [ARTConnectionRecoveryKey fromJsonString:options.recover];
+            self.msgSerial = recoveryKey.msgSerial;
+            self.connection.latestMessageSerial = self.msgSerial;
+            [self recoverChannels: recoveryKey]; //RTN16j
+        } else {
+            self.msgSerial = 0;
+            self.connection.latestMessageSerial = 0;
+        }
     }
     return self;
 }
 
+- (void)recoverChannels:(ARTConnectionRecoveryKey *)recoveryKey {
```

Comment:
A more Objective-C idiomatic method selector would be `recoverChannelsWithRecoveryKey:`. Also, you can make the `recoverKey` pointer `const`.